packages/scope-manager/src/analyze.ts

```diff
@@ -122,7 +95,7 @@ function analyze(
     jsxFragmentName:
       providedOptions?.jsxFragmentName ?? DEFAULT_OPTIONS.jsxFragmentName,
     sourceType: providedOptions?.sourceType ?? DEFAULT_OPTIONS.sourceType,
-    lib: providedOptions?.lib ?? [mapEcmaVersion(ecmaVersion)],
```

Comment:
Hmmm. I wonder if maybe I was too over-zealous with my recommendation to cull higher?

Having this auto-resolved is a really nice property. For the most part it shouldn't cause any issues for people (it just includes a few more global variables for people using rules like `no-undef` or `no-use-before-define`), but there might be edge cases?

Hmm. WDYT? I'm torn on if it's worth the code.

Reply:
Neither of those are rules we recommend folks use - and folks should generally always be in >=ES2021 or so these days. I say let's kill them for now, and use the RC period to see if anybody notices! 🔪

Reply:
Oh I will mark this as a breaking change now though.